components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -955,9 +967,33 @@ void updateConsumerApplication(OAuthConsumerAppDTO consumerAppDTO, boolean enabl
             oAuthAppDO.setRequestObjectEncryptionMethod(requestObjectEncryptionMethod);
             oAuthAppDO.setRequirePushedAuthorizationRequests(consumerAppDTO.getRequirePushedAuthorizationRequests());
             oAuthAppDO.setSubjectTokenEnabled(consumerAppDTO.isSubjectTokenEnabled());
-            oAuthAppDO.setSubjectTokenExpiryTime(consumerAppDTO.getSubjectTokenExpiryTime());;
+            oAuthAppDO.setSubjectTokenExpiryTime(consumerAppDTO.getSubjectTokenExpiryTime());
+            if (OAuth2Util.isJWTAccessTokenOIDCClaimsSeparationEnabled() &&
+                    oAuthAppDO.isJwtAccessTokenOIDCClaimsSeparationEnabled()) {
+                validateJwtAccessTokenClaims(consumerAppDTO, tenantDomain);
+                oAuthAppDO.setJwtAccessTokenClaims(consumerAppDTO.getJwtAccessTokenOIDCClaims());
+            }
         }
         dao.updateConsumerApplication(oAuthAppDO);
+        // We only trigger the jwt access token claims migration if the following conditions are met.
+        // 1. The new feature is enabled.
+        // 2. The app is not already in the new feature.
```

Comment:
Please re-phrase the second line of this comment.

Reply:
fixed